pkg/agent/clean/adunmigration/rtbs.go

```diff
@@ -390,6 +465,27 @@ func migrateGRBs(workunit *migrateUserWorkUnit, sc *config.ScaledContext, dryRun
 				}
 				return true, nil
 			})
+			if err != nil {
+				logrus.Errorf("[%v] permanent error when deleting GRB, giving up: %v", migrateGrbsOperation, err)
+				continue
+			}
+
+			// Adding the annotation here has the side effect of triggering a downstream RBAC sync for our newly created
+			// binding, which is why we perform it last
+			newAnnotations[adGUIDMigrationAnnotation] = oldGrb.UserName
+			err = wait.ExponentialBackoff(backoff, func() (finished bool, err error) {
+				updatedGrb, err := sc.Management.GlobalRoleBindings("").GetNamespaced(oldGrb.Namespace, createdGrb.Name, metav1.GetOptions{})
+				updatedGrb.Annotations = newAnnotations
+				_, err = sc.Management.GlobalRoleBindings("").Update(updatedGrb)
+				if err != nil {
+					if apierrors.IsInternalError(err) {
+						logrus.Errorf("[%v] internal error while deleting GRB, will backoff and retry: %v", migrateGrbsOperation, err)
```

Comment:
Log message needs to be updated to state the GRB update operation and not delete.
"internal error while updating GRB"